File: backend/src/hono/routes/r2/chunk-naming.ts
export const R2_CHUNK_PREFIX = 'chunk-';
export const R2_CHUNK_VERSION_SEPARATOR = '-v';
export const R2_CHUNK_EXTENSION = '.dat';

export function getChunkKey(version, chunkName) {
  return `${R2_CHUNK_PREFIX}${chunkName}${R2_CHUNK_VERSION_SEPARATOR}${version}${R2_CHUNK_EXTENSION}`;
}

File: lib/features/r2/chunk_naming.dart
import 'package:flutter/material.dart';

class R2ChunkConfig {
  static const String chunkPrefix = 'chunk-';
  static const String chunkVersionSeparator = '-v';
  static const String chunkExtension = '.dat';

  static String getChunkKey(String version, String chunkName) {
    return '$chunkPrefix$chunkName$chunkVersionSeparator$version$chunkExtension';
  }
}
